openhands/runtime/impl/eventstream/eventstream_runtime.py

```diff
@@ -190,6 +191,14 @@ def __init__(
                 f'Installing extra user-provided dependencies in the runtime image: {self.config.sandbox.runtime_extra_deps}',
             )
 
+        # if docker context endpoint is set, custom sandbox is assumed.
+        if self.config.sandbox.docker_endpoint:
```

Comment:
this seems like an unnecessary assumption--isn't it possible someone would want to set the docker endpoint _without_ this behavior?

Reply:
Not as is currently implemented. This is a new configuration attribute.

Reply:
A good point that docker_endpoint can be used and managed by openhands in future for remote instances.
Instroducing a new config attribute `attach_to_existing` to make it explicit.